src/map/mob.c

```diff
@@ -312,6 +312,46 @@ static struct view_data *mob_get_viewdata(int class_)
 		return 0;
 	return &mob->db(class_)->vd;
 }
+
+/**
+ * Create unique view data associated to a spawned monster.
+ * @param md: Mob to adjust
+ */
+void mob_set_dynamic_viewdata(struct mob_data* md) {
+	// If it is a valid monster and it has not already been created
+	if (md && !md->vd_changed) {
+		// Allocate a dynamic entry
+		struct view_data* vd = (struct view_data*)aMalloc(sizeof(struct view_data));
+
+		// Copy the current values
+		memcpy(vd, md->vd, sizeof(struct view_data));
+
+		// Update the pointer to the new entry
+		md->vd = vd;
+
+		// Flag it as changed so it is freed later on
+		md->vd_changed = true;
+	}
+}
+
+/**
+ * Free any view data associated to a spawned monster.
+ * @param md: Mob to free
+ */
+void mob_free_dynamic_viewdata(struct mob_data* md) {
+	// If it is a valid monster and it has already been allocated
+	if (md && md->vd_changed) {
```

Comment:
md != NULL

You can also reduce the indentation level here by doing the opposite and returning.

Reply:
Uploaded changes